### ISSUE TITLE
vcrun2010 installation workaround (errors possibly caused by sigsegv's in msxml3 installer)

### ISSUE THREAD

**Issue by WheezyE** [ptitSeb/box86#240]
`winetricks -q vcrun2010` installation fails with: 
```
pi@raspberrypi:~ $ winetricks vcrun2010
------------------------------------------------------
Your version of wine Box86
5.18 is no longer supported upstream. You should upgrade to 4.x
------------------------------------------------------
Executing mkdir -p /home/pi
Using winetricks 20200412 - sha256sum: a7b3348ebebc44bf7c0068cae5ee37e2c0963ca5afc27196367a882b4839a0b6 with Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
wine-5.18 and WINEARCH=win32
Executing w_do_call vcrun2010
Executing mkdir -p /home/pi
Executing load_vcrun2010 
Using native,builtin override for following DLLs: msvcp100 msvcr100 vcomp100 atl100
Executing wine regedit C:\windows\Temp\override-dll.reg
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
0050:err:vulkan:wine_vk_init Failed to load libvulkan.so.1.
0050:err:ole:start_rpcss Failed to start RpcSs service
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Executing cd /home/pi/.cache/winetricks/vcrun2010
Executing wine vcredist_x86.exe
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
00c8:fixme:clusapi:GetNodeClusterState ((null),7F29EC44) stub!
00c8:fixme:advapi:DecryptFileA ("y:\\ea58d1d9a0191cae9b1493\\", 00000000): stub
00dc:err:winediag:load_gssapi_krb5 Failed to load libgssapi_krb5, Kerberos SSP support will not be available.
00dc:fixme:ntdll:EtwRegisterTraceGuidsW (00545F38, 00550180, {e2821408-c59d-418f-ad3f-aa4e792aeb79}, 1, 7F5CE9B0, (null), (null), 00550188): stub
00dc:fixme:ntdll:EtwRegisterTraceGuidsW   register trace class {e2821408-c59d-418f-ad3f-aa4e792aeb79}
00dc:fixme:heap:RtlSetHeapInformation 00000000 1 00000000 0 stub
00dc:fixme:heap:RtlSetHeapInformation 00000000 1 00000000 0 stub
00dc:err:module:open_builtin_file failed to load .so lib "/home/pi/wine/lib/wine/msxml3.dll.so"
00dc:err:ole:apartment_add_dll couldn't load in-process dll L"C:\\windows\\system32\\msxml3.dll"
00dc:err:ole:create_server class {f5078f32-c551-11d3-89b9-0000f81fe221} not registered
00dc:fixme:ole:com_get_class_object CLSCTX_REMOTE_SERVER not supported
00dc:err:ole:com_get_class_object no class object {f5078f32-c551-11d3-89b9-0000f81fe221} could be created for context 0x17
00dc:err:module:open_builtin_file failed to load .so lib "/home/pi/wine/lib/wine/msxml3.dll.so"
00dc:err:ole:apartment_add_dll couldn't load in-process dll L"C:\\windows\\system32\\msxml3.dll"
00dc:err:ole:create_server class {f5078f32-c551-11d3-89b9-0000f81fe221} not registered
00dc:fixme:ole:com_get_class_object CLSCTX_REMOTE_SERVER not supported
00dc:err:ole:com_get_class_object no class object {f5078f32-c551-11d3-89b9-0000f81fe221} could be created for context 0x17
00dc:err:module:open_builtin_file failed to load .so lib "/home/pi/wine/lib/wine/msxml3.dll.so"
00dc:err:ole:apartment_add_dll couldn't load in-process dll L"C:\\windows\\system32\\msxml3.dll"
00dc:err:ole:create_server class {f5078f32-c551-11d3-89b9-0000f81fe221} not registered
00dc:fixme:ole:com_get_class_object CLSCTX_REMOTE_SERVER not supported
00dc:err:ole:com_get_class_object no class object {f5078f32-c551-11d3-89b9-0000f81fe221} could be created for context 0x17
00dc:fixme:advapi:LsaOpenPolicy ((null),7F5CEC78,0x00000001,7F5CEC6C) stub
00dc:fixme:security:GetWindowsAccountDomainSid (7F5CEBF8 004AC4C4 7F5CEBF4): semi-stub
00dc:fixme:secur32:GetComputerObjectNameW NameFormat 7 not implemented
00dc:fixme:advapi:LsaClose (0000CAFE) stub
00dc:fixme:wer:WerReportCreate build report information from scratch for 004AC4F8
00dc:fixme:wer:WerReportAddDump (004AC4F8, 00000094, 00000000, 2, 7F5CE5B8, 00000000, 0) :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 0, (null), L"Setup") :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 1, (null), L"Unknown") :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 2, (null), L"10.0.30319.1") :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 3, (null), L"1") :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 4, (null), L" ") :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 5, (null), L"None_UI_Interactive_Crash") :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 6, (null), L"0xe06d7363") :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 7, (null), L"0") :stub
00dc:fixme:wer:WerReportSetParameter (004AC4F8, 8, (null), L" ") :stub
00dc:fixme:wer:WerReportAddFile (004AC4F8, L"C:\\users\\pi\\Temp\\Setup_20201025_141557465.html", 5, 0x0) :stub
00dc:fixme:wer:WerReportSubmit (004AC4F8, 1, 0x10, 7F5CE5C4) :stub
00dc:fixme:ntdll:EtwUnregisterTraceGuids deadbeef: stub
------------------------------------------------------
Note: command wine vcredist_x86.exe returned status 1. Aborting.
------------------------------------------------------
pi@raspberrypi:~ $ 009c:err:rpc:I_RpcReceive we got fault packet with status 0x1c010003
```

But I can get around the errors with the following :)
- 1. `winetricks msxml3` (also fails, but does override wine dll's like its supposed to and extract the msxml3.msi file it downloads). 
```
pi@raspberrypi:~ $ winetricks msxml3
------------------------------------------------------
Your version of wine Box86
5.18 is no longer supported upstream. You should upgrade to 4.x
------------------------------------------------------
Executing mkdir -p /home/pi
Using winetricks 20200412 - sha256sum: a7b3348ebebc44bf7c0068cae5ee37e2c0963ca5afc27196367a882b4839a0b6 with Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
wine-5.18 and WINEARCH=win32
Executing w_do_call msxml3
Executing mkdir -p /home/pi
Executing load_msxml3 
Using native override for following DLLs: msxml3
Executing wine regedit C:\windows\Temp\override-dll.reg
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Executing cd /home/pi/.cache/winetricks/msxml3
------------------------------------------------------
Working around wine bug 26925 -- Forcing quiet install
------------------------------------------------------
Executing wine msiexec /i msxml3.msi /q
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
0218:fixme:file:NtLockFile I/O completion on lock not implemented yet
0218:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0280:fixme:setupapi:SetupQueryInfOriginalFileInformationW (793D0F58, 0, 00000000, 7BAECFC0): semi-stub
0280:fixme:setupapi:SetupPromptReboot 004A6A50, 00000000, 1: stub
24625|Double SIGSEGV!
24625|Double SIGSEGV!
Segmentation fault
------------------------------------------------------
Note: command wine msiexec /i msxml3.msi /q returned status 139. Aborting.
------------------------------------------------------
```
- 2. Manually complete the msxml3 install with `cp ~/.cache/winetricks/msxml3/%SystemDrive%/Program\ Files/Common\ Files/Microsoft\ Shared/SFPCA\ Cache/* ~/.wine/drive_c/windows/system32/`.   If we don't do this step, the vcrun2010 installer seems to crash before we can click the "Finish" button.

- 3. Then we can do `winetricks vcrun2010` or  `winetricks -q vcrun2010` and it should install, though with some errors(?).  I'm not sure the integrity of the install yet, but wanted to post here for anybody trying to install vcrun2010

```
pi@raspberrypi:~ $ winetricks --force vcrun2010
------------------------------------------------------
Your version of wine Box86
5.18 is no longer supported upstream. You should upgrade to 4.x
------------------------------------------------------
Executing mkdir -p /home/pi
Using winetricks 20200412 - sha256sum: a7b3348ebebc44bf7c0068cae5ee37e2c0963ca5afc27196367a882b4839a0b6 with Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
wine-5.18 and WINEARCH=win32
Executing w_do_call vcrun2010
Executing mkdir -p /home/pi
Executing load_vcrun2010 
Using native,builtin override for following DLLs: msvcp100 msvcr100 vcomp100 atl100
Executing wine regedit C:\windows\Temp\override-dll.reg
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Executing cd /home/pi/.cache/winetricks/vcrun2010
Executing wine vcredist_x86.exe
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
0154:fixme:clusapi:GetNodeClusterState ((null),7F29EC44) stub!
0154:fixme:advapi:DecryptFileA ("y:\\1b252340160487ceaad07776\\", 00000000): stub
0168:err:winediag:load_gssapi_krb5 Failed to load libgssapi_krb5, Kerberos SSP support will not be available.
0168:fixme:ntdll:EtwRegisterTraceGuidsW (00545F38, 00550180, {e2821408-c59d-418f-ad3f-aa4e792aeb79}, 1, 7F5CE9B0, (null), (null), 00550188): stub
0168:fixme:ntdll:EtwRegisterTraceGuidsW   register trace class {e2821408-c59d-418f-ad3f-aa4e792aeb79}
0168:fixme:heap:RtlSetHeapInformation 00000000 1 00000000 0 stub
0168:fixme:heap:RtlSetHeapInformation 00000000 1 00000000 0 stub
0168:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
0168:fixme:advapi:LsaOpenPolicy ((null),7F5CF518,0x00000001,7F5CF50C) stub
0168:fixme:security:GetWindowsAccountDomainSid (7F5CF498 004C7244 7F5CF494): semi-stub
0168:fixme:secur32:GetComputerObjectNameW NameFormat 7 not implemented
0168:fixme:advapi:LsaClose (0000CAFE) stub
0168:err:ole:apartment_getclassobject DllGetClassObject returned error 0x80040111 for dll L"C:\\windows\\system32\\msxml6.dll"
0168:err:ole:create_server class {88d96a05-f192-11d4-a65f-0040963251e5} not registered
0168:fixme:ole:com_get_class_object CLSCTX_REMOTE_SERVER not supported
0168:err:ole:com_get_class_object no class object {88d96a05-f192-11d4-a65f-0040963251e5} could be created for context 0x17
019c:fixme:wintrust:SOFTPUB_VerifyImageHash Cannot verify hash for pszObjId="1.3.6.1.4.1.311.2.1.25"
019c:fixme:file:NtLockFile I/O completion on lock not implemented yet
019c:fixme:wintrust:SOFTPUB_VerifyImageHash Cannot verify hash for pszObjId="1.3.6.1.4.1.311.2.1.30"
0168:fixme:advapi:LsaOpenPolicy ((null),7F5CF368,0x00000001,7F5CF35C) stub
0168:fixme:security:GetWindowsAccountDomainSid (7F5CF2E8 B1164F9C 7F5CF2E4): semi-stub
0168:fixme:secur32:GetComputerObjectNameW NameFormat 7 not implemented
0168:fixme:advapi:LsaClose (0000CAFE) stub
0168:fixme:ntdll:EtwUnregisterTraceGuids deadbeef: stub
```

Side-note: we can get around the `0168:err:ole:apartment_getclassobject DllGetClassObject returned error 0x80040111 for dll L"C:\\windows\\system32\\msxml6.dll"` with `winetricks msxml6`, but I'm not sure if this step is needed since we still get a similar output from vcrun2010:
```
pi@raspberrypi:~ $ winetricks --force vcrun2010
------------------------------------------------------
Your version of wine Box86
5.18 is no longer supported upstream. You should upgrade to 4.x
------------------------------------------------------
Executing mkdir -p /home/pi
Using winetricks 20200412 - sha256sum: a7b3348ebebc44bf7c0068cae5ee37e2c0963ca5afc27196367a882b4839a0b6 with Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
wine-5.18 and WINEARCH=win32
Executing w_do_call vcrun2010
Executing mkdir -p /home/pi
Executing load_vcrun2010 
Using native,builtin override for following DLLs: msvcp100 msvcr100 vcomp100 atl100
Executing wine regedit C:\windows\Temp\override-dll.reg
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Executing cd /home/pi/.cache/winetricks/vcrun2010
Executing wine vcredist_x86.exe
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
Box86 with Dynarec v0.1.4 0edc0374 built on Oct 24 2020 05:29:33
0158:fixme:clusapi:GetNodeClusterState ((null),7F29EC44) stub!
0158:fixme:advapi:DecryptFileA ("y:\\efea7248d49648ac0573f76f5b59093c\\", 00000000): stub
016c:err:winediag:load_gssapi_krb5 Failed to load libgssapi_krb5, Kerberos SSP support will not be available.
016c:fixme:ntdll:EtwRegisterTraceGuidsW (00545F38, 00550180, {e2821408-c59d-418f-ad3f-aa4e792aeb79}, 1, 7F5CE9B0, (null), (null), 00550188): stub
016c:fixme:ntdll:EtwRegisterTraceGuidsW   register trace class {e2821408-c59d-418f-ad3f-aa4e792aeb79}
016c:fixme:heap:RtlSetHeapInformation 00000000 1 00000000 0 stub
016c:fixme:heap:RtlSetHeapInformation 00000000 1 00000000 0 stub
016c:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
016c:fixme:advapi:LsaOpenPolicy ((null),7F5CF518,0x00000001,7F5CF50C) stub
016c:fixme:security:GetWindowsAccountDomainSid (7F5CF498 004C728C 7F5CF494): semi-stub
016c:fixme:secur32:GetComputerObjectNameW NameFormat 7 not implemented
016c:fixme:advapi:LsaClose (0000CAFE) stub
0170:fixme:wintrust:SOFTPUB_VerifyImageHash Cannot verify hash for pszObjId="1.3.6.1.4.1.311.2.1.25"
0170:fixme:file:NtLockFile I/O completion on lock not implemented yet
0170:fixme:wintrust:SOFTPUB_VerifyImageHash Cannot verify hash for pszObjId="1.3.6.1.4.1.311.2.1.30"
016c:fixme:advapi:LsaOpenPolicy ((null),7F5CF368,0x00000001,7F5CF35C) stub
016c:fixme:security:GetWindowsAccountDomainSid (7F5CF2E8 B10D01E4 7F5CF2E4): semi-stub
016c:fixme:secur32:GetComputerObjectNameW NameFormat 7 not implemented
016c:fixme:advapi:LsaClose (0000CAFE) stub
016c:fixme:ntdll:EtwUnregisterTraceGuids deadbeef: stub
```

**Comment by retropiepro** [comment]
@WheezyE How did you get winetricks to work?

When I run it on my pi I get this error:

```
Executing mkdir -p /home/pi
Using winetricks 20200412-next - sha256sum: 8e180b094e9cde430a89ee98ca63855bfacf3479b030b9bd4fe24bbd261ee04a with Box86 with Dynarec v0.1.5 304eaab4 built on Nov  1 2020 18:42:27
wine-5.20 and WINEARCH=win32
installed_file1="Box86 with Dynarec v0.1.5 304eaab4 built on Nov  1 2020 18:42:27
------------------------------------------------------
warning: bug: w_metadata adobeair corrupt, might need forward slashes?
------------------------------------------------------
```
Thanks

**Comment by ptitSeb** [comment]
use `BOX86_NOBANNER=1` env. var. to silent box86, and winetricks will work properly.

**Comment by retropiepro** [comment]
Thanks @ptitSeb you are awesome!

**Comment by WheezyE** [comment]
This is fixed now with commit 0f2998a672851eb2439a77ac200973b7d50ddbac  I'll add vcrun2010 to the compatibility list
Thanks Seb!
